packages/components/tree-v2/src/hooks/useCheck.ts

```diff
@@ -0,0 +1,220 @@
+import { nextTick, ref, watch } from 'vue'
+import { NODE_CHECK_CHANGE, NODE_CHECK } from '../emits'
+import type { Ref } from 'vue'
+import type {
+  ITreeProps,
+  TreeKey,
+  TreeNode,
+  Tree,
+  TreeNodeData,
+} from '../tree.type'
+
+const DELETE = 'delete'
+const ADD = 'add'
+
+export function useCheck(props: ITreeProps, tree: Ref<Tree>, emit) {
+  const checkedKeys = ref<Set<TreeKey>>(new Set())
+  const indeterminateKeys = ref<Set<TreeKey>>(new Set())
+
+  watch(
+    () => tree.value,
+    () => {
+      return nextTick(() => {
+        _setCheckedKeys(props.defaultCheckedKeys)
+      })
```

Comment:
I think `return` here is unnecessary since there will be no handler for your return.